### ISSUE TITLE
IndexOutOfBounds-Exception

### ISSUE THREAD

**Issue by AlexSchuetz** [ChasakisD/XFExpandableListView#4]
Hi,

I am using version 1.0.0 and as soon as I have one ExpandableGroup in my ObservableCollection I get a IndexOutOfBoundsException.

NewInstance has been called for every ExpandableGroup in the collection.

Any ideas?

**Comment by ChasakisD** [comment]
Hi,

Could you post a reproduction solution or how you use the library?

**Comment by AlexSchuetz** [comment]
Unfortunately I cannot share my code as it is closed source. I will try to modify the sample project to reproduce the error.

**Comment by AlexSchuetz** [comment]
I found a way to reproduce it. I pushed to my [fork](https://github.com/AlexSchuetz/XFExpandableListView).

I may not add a view containing the ExpandableList to my page.
I hope I am not missing something here, since I am pretty new to Xamarin.Forms

**Comment by ChasakisD** [comment]
I can reproduce the exception. It is thrown on the ListViewRenderer and it is happening on the ```line 112``` of the ```ExpandableListView``` when trying to set the ItemsSource of the ListView. The full stacktrace is 
```
09-29 16:28:36.071 I/MonoDroid( 2968): UNHANDLED EXCEPTION:
09-29 16:28:36.111 I/MonoDroid( 2968): System.ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
09-29 16:28:36.111 I/MonoDroid( 2968): Parameter name: index
09-29 16:28:36.111 I/MonoDroid( 2968):   at System.Collections.Generic.List`1[T].get_Item (System.Int32 index) [0x00009] in <43dbbdc147f2482093d8409abb04c233>:0 
09-29 16:28:36.111 I/MonoDroid( 2968):   at Cadenza.Collections.OrderedDictionary`2[TKey,TValue].get_Item (System.Int32 index) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\OrderedDictionary.cs:100 
09-29 16:28:36.111 I/MonoDroid( 2968):   at Xamarin.Forms.Internals.TemplatedItemsList`2[TView,TItem].GetGroup (System.Int32 index) [0x0000a] in D:\a\1\s\Xamarin.Forms.Core\TemplatedItemsList.cs:569 
09-29 16:28:36.111 I/MonoDroid( 2968):   at Xamarin.Forms.Internals.TemplatedItemsList`2[TView,TItem].Xamarin.Forms.ITemplatedItemsList<TItem>.GetGroup (System.Int32 index) [0x00000] in D:\a\1\s\Xamarin.Forms.Core\TemplatedItemsList.cs:574 
09-29 16:28:36.111 I/MonoDroid( 2968):   at Xamarin.Forms.Platform.Android.ListViewAdapter.get_Count () [0x0002d] in D:\a\1\s\Xamarin.Forms.Platform.Android\Renderers\ListViewAdapter.cs:85 
09-29 16:28:36.111 I/MonoDroid( 2968):   at Android.Widget.BaseAdapter.n_GetCount (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <a10f61e70eeb434e952fef884856c199>:0 
09-29 16:28:36.111 I/MonoDroid( 2968):   at (wrapper dynamic-method) System.Object.46(intptr,intptr)
```

The weird issue is that if I take the ExpandableListView just copy and paste it to the ExpandableListViewPage, no exception is thrown. This is so weird! Maybe something is messing up with the bindingcontext (the same exception is thrown if you set the bindingcontext from xaml) and how it is updates the adapter? When i'll have more time i will investigate it further. Did this also happens on iOS? Have you tested it? or @GregoryMaks could? 

I will update here as soon i have more information.

**Comment by GregoryMaks** [comment]
@AlexSchuetz @ChasakisD I have a few things to push as a PR for stability purposes. Probably it will address this issue. I will check in a few hours to see what's happening. There will definitely be a new PR after that.

**Comment by AlexSchuetz** [comment]
I will test it tomorrow or at least on monday.

**Comment by AlexSchuetz** [comment]
@GregoryMaks Can you issue a pull request to my fork, too? Perhaps the issue is fixed then.

**Comment by AlexSchuetz** [comment]
IOS produces the same Exception. I will check in an iOS-Sample project on my fork, I will also add a UWP-project.

**Comment by GregoryMaks** [comment]
@AlexSchuetz @ChasakisD I pushed a PR with some fixes, please take a look: https://github.com/ChasakisD/XFExpandableListView/pull/5.

@AlexSchuetz The issue mentioned above is not reproducing after the fix.

**Comment by ChasakisD** [comment]
Issue was fixed from #5
